api/web3server.go

```diff
@@ -433,7 +435,24 @@ func (svr *web3Handler) sendRawTransaction(in *gjson.Result) (interface{}, error
 		return nil, errInvalidFormat
 	}
 	// parse raw data string from json request
-	tx, sig, pubkey, err := action.DecodeRawTx(dataStr.String(), svr.coreService.EVMNetworkID())
+	var (
+		tx       *types.Transaction
+		sig      []byte
+		pubkey   crypto.PublicKey
+		err      error
+		encoding iotextypes.Encoding
+		cs       = svr.coreService
+	)
+	if g := cs.Genesis(); g.IsSumatra(cs.TipHeight()) {
+		tx, encoding, sig, pubkey, err = action.DecodeEtherTx(dataStr.String())
```

Comment:
Don't see the value of wrapping the logic in a public function in package `action`, which is only called here.

Reply:
2 reasons:
1. the current code is `DecodeRawTx` (which only works for EIP-155), so put this func together with current code
2. this func is to deal with Eth-compatible tx decoding/conversion, so it makes more sense to put it (and all related codes) in the `action` package. I think it's better and clear to have all these codes in 1 place